fedimint-core/src/api.rs

```diff
@@ -410,6 +412,22 @@ where
     })
 }
 
+lazy_static! {
+    /// Small LRU used as `GlobalFederatinApi::await_block` cache.
+    ///
+    /// This is mostly to avoid multiple client module recovery processes
+    /// re-requesting same blocks and putting burden on the federation.
+    ///
+    /// The LRU can be be fairly small, as if the modules are (near-)bottlenecked
+    /// on fetching blocks they will naturally synchronize, or split into
+    /// a handful of groups. And if they are not, no LRU here is going to help
+    /// them.
+    static ref API_AWAIT_BLOCK_LRU: tokio::sync::Mutex<lru::LruCache<u64, Arc<tokio::sync::Mutex<Option<SessionOutcome>>>>> =
+        tokio::sync::Mutex::new(lru::LruCache::new(
+            NonZeroUsize::new(32).expect("is non-zero")
+        ));
+}
+
```

Comment:
If this *has to be* global then the federation id has to be part of the cache key, otherwise two clients running in the same process would get blocks from each other's federations.

Also, didn't we rename blocks to session outcomes? @joschisan

Reply:
yes we did in the api endpoints and the struct itself, just not everywhere in the code.

Reply:
> If this _has to be_ global then the federation id has to be part of the cache key, otherwise two clients running in the same process would get blocks from each other's federations.

:facepalm: Right, good point.



> Also, didn't we rename blocks to session outcomes?

So I'm supposed to call it "sessions" everywhere from now on instead of blocks?

Reply:
> So I'm supposed to call it "sessions" everywhere from now on instead of blocks?

Yes, sorry for the confusion. We renamed it (again) after you (and me too) complained about the ambiguity.